sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangeInternal.java

```diff
@@ -88,32 +83,36 @@ public String toJsonString() {
         return this.toJson();
     }
 
-    public static FeedRangeInternal tryParse(final String jsonString) throws IOException {
-        if (jsonString == null) {
-            throw new NullPointerException("jsonString");
-        }
-
+    public static FeedRangeInternal tryParse(final String jsonString) {
+        checkNotNull(jsonString, "Argument 'jsonString' must not be null");
         final ObjectMapper mapper = Utils.getSimpleObjectMapper();
-        JsonNode rootNode = mapper.readTree(jsonString);
-
-        JsonNode rangeNode = rootNode.get(Constants.Properties.RANGE);
-        if (rangeNode != null && rangeNode.isObject()) {
-            Range<String> range = new Range<>((ObjectNode)rangeNode);
-            return new FeedRangeEpkImpl(range);
-        }
 
-        JsonNode pkNode = rootNode.get(Constants.Properties.FEED_RANGE_PARTITION_KEY);
-        if (pkNode != null && pkNode.isArray()) {
-            PartitionKeyInternal pk = mapper.convertValue(pkNode, PartitionKeyInternal.class);
-            return new FeedRangePartitionKeyImpl(pk);
-        }
-
-        JsonNode pkRangeIdNode =
-            rootNode.get(Constants.Properties.FEED_RANGE_PARTITION_KEY_RANGE_ID);
-        if (pkRangeIdNode != null && pkRangeIdNode.isTextual()) {
-            return new FeedRangePartitionKeyRangeImpl(pkRangeIdNode.asText());
+        try {
+            JsonNode rootNode = mapper.readTree(jsonString);
+
+            JsonNode rangeNode = rootNode.get(Constants.Properties.RANGE);
+            if (rangeNode != null && rangeNode.isObject()) {
+                Range<String> range = new Range<>((ObjectNode)rangeNode);
+                return new FeedRangeEpkImpl(range);
+            }
+
+            JsonNode pkNode = rootNode.get(Constants.Properties.FEED_RANGE_PARTITION_KEY);
+            if (pkNode != null && pkNode.isArray()) {
+                PartitionKeyInternal pk = mapper.convertValue(pkNode, PartitionKeyInternal.class);
+                return new FeedRangePartitionKeyImpl(pk);
+            }
+
+            JsonNode pkRangeIdNode =
+                rootNode.get(Constants.Properties.FEED_RANGE_PARTITION_KEY_RANGE_ID);
+            if (pkRangeIdNode != null && pkRangeIdNode.isTextual()) {
+                return new FeedRangePartitionKeyRangeImpl(pkRangeIdNode.asText());
+            }
+
+            return null;
+
+        } catch (final IOException ioError) {
+            LOGGER.debug("Failed to parse feed range JSON %s", jsonString, ioError);
```

Comment:
use slf4j logger placeholder instead of `%s`
```suggestion
            LOGGER.debug("Failed to parse feed range JSON {}", jsonString, ioError);
```